gst/nnstreamer/tensor_source/tensor_src_iio.c

```diff
@@ -260,15 +757,274 @@ gst_tensor_src_iio_finalize (GObject * object)
+  if (fd == NULL) {
+    GST_ERROR ("Unable to open file to write %s.\n", filename);
+    goto error_free_filename;
+  }
+
+  bytes_printed = fprintf (fd, "%s", contents);
+  if (bytes_printed != strlen (contents)) {
+    GST_ERROR ("Unable to write to file %s.\n", filename);
+    goto error_close_file;
+  }
+  if (!fclose (fd)) {
+    GST_ERROR ("Unable to close file %s after write.\n", filename);
+    goto error_free_filename;
+  }
+  ret = TRUE;
+
+  if (DBG) {
+    gchar *file_contents = NULL;
+    ret = FALSE;
+    if (!g_file_get_contents (filename, &file_contents, NULL, &error)) {
+      GST_ERROR ("Unable to read file %s with error %s.\n", filename,
+          error->message);
+      goto error_free_filename;
+    } else {
+      if (!g_strcmp0 (contents, file_contents)) {
+        ret = TRUE;
+      }
+      g_free (file_contents);
+    }
+  }
+
+  g_free (filename);
+  return ret;
+
+error_close_file:
+  fclose (fd);
+
+error_free_filename:
+  g_error_free (error);
+  g_free (filename);
+  return ret;
+}
+
+/**
+ * @brief write the int in to the file
+ * @param[in] self Tensor src IIO object
+ * @param[in] file Destination file for the data
+ * @param[in] base_dir Directory containing the file
+ * @param[in] contents Data to be written to the file
+ * @return True if write was successful, false on failure
+ */
+static gboolean
+gst_tensor_write_sysfs_int (GstTensorSrcIIO * self, const gchar * file,
+    const gchar * base_dir, const gint contents)
+{
+  gchar *contents_char = NULL;
+  gboolean ret;
+
+  contents_char = g_strdup_printf ("%d", contents);
+  ret = gst_tensor_write_sysfs_string (self, file, base_dir, contents_char);
+
+  g_free (contents_char);
+  return ret;
+}
+
+/**
+ * @brief set value to all the channels
+ * @param[in] self Tensor src IIO object
+ * @param[in] contents Data to be written to the file
+ * @return True if write was successful, false if failure on any channel
+ */
+static gboolean
+gst_tensor_set_all_channels (GstTensorSrcIIO * self, const gint contents)
+{
+  GList *ch_list;
+  gchar *filename = NULL;
+  GstTensorSrcIIOChannelProperties *channel_prop = NULL;
+  gboolean ret = TRUE;
+
+  for (ch_list = self->channels; ch_list != NULL; ch_list = ch_list->next) {
+    channel_prop = (GstTensorSrcIIOChannelProperties *) ch_list->data;
+    filename = g_strdup_printf ("%s%s", channel_prop->name, EN_SUFFIX);
+    ret &=
+        gst_tensor_write_sysfs_int (self, filename, channel_prop->base_dir,
+        contents);
+    g_free (filename);
+  }
+
+  return ret;
+}
+
 /**
  * @brief start function, called when state changed null to ready.
+ * load the device and init the device resources
  */
 static gboolean
 gst_tensor_src_iio_start (GstBaseSrc * src)
 {
   /** load and init resources */
-  //FIXME: fill this function
+  GstTensorSrcIIO *self;
+  self = GST_TENSOR_SRC_IIO_CAST (src);
+  gint id;
+  guint64 sampling_frequency;
+  gchar *dirname = NULL;
+  gchar *filename = NULL;
+
+  // Find the device
```

Comment:
Now meson default option 'gnu89'  does not make an error for line comment with  //.
Please change all to /* */ in next PR.

Reply:
Noted, will update in next PR.